staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -96,171 +97,176 @@ func isInterfaceUp(intf *net.Interface) bool {
 	return false
 }
 
-//getFinalIP method receives all the IP addrs of a Interface
-//and returns a nil if the address is Loopback, Ipv6, link-local or nil.
-//It returns a valid IPv4 if an Ipv4 address is found in the array.
-func getFinalIP(addrs []net.Addr) (net.IP, error) {
+func isLoopbackOrPointToPoint(intf *net.Interface) bool {
+	if intf.Flags&(net.FlagLoopback|net.FlagPointToPoint) != 0 {
+		return true
+	}
+	return false
+}
+
+// findGlobalIP returns the first IP on interface that is not a loopback or
+// link local address. Can optionally allow IPv6 addresses. Note: ordering
+// of addresses processed is undefined.
+func findGlobalIP(addrs []net.Addr, allowIPv6 bool) (net.IP, error) {
 	if len(addrs) > 0 {
-		for i := range addrs {
-			glog.V(4).Infof("Checking addr  %s.", addrs[i].String())
-			ip, _, err := net.ParseCIDR(addrs[i].String())
+		for _, a := range addrs {
+			glog.V(4).Infof("Checking address %q", a.String())
+			ip, _, err := net.ParseCIDR(a.String())
 			if err != nil {
 				return nil, err
 			}
-			//Only IPv4
-			//TODO : add IPv6 support
-			if ip.To4() != nil {
-				if !ip.IsLoopback() && !ip.IsLinkLocalMulticast() && !ip.IsLinkLocalUnicast() {
-					glog.V(4).Infof("IP found %v", ip)
-					return ip, nil
-				} else {
-					glog.V(4).Infof("Loopback/link-local found %v", ip)
-				}
-			} else {
-				glog.V(4).Infof("%v is not a valid IPv4 address", ip)
+			if ip.To4() == nil && !allowIPv6 {
+				glog.V(4).Infof("Ignoring IPv6 address '%v' for IPv4 only mode", ip)
+				continue
 			}
-
+			// TODO: Do we need to test IsLinkLocalMulticast or IsInterfaceLocalMulticast?
+			// net.Addrs() indicates that it returns *unicast* addresses (and not multicast).
```

Comment:
@leblancd Nice. That would be a better method to use. The comment was intended to be removed, once a decision was made as to whether we should (like the old code) test for MCAST addresses.